core/json/src/main/scala/net/liftweb/json/JsonParser.scala

```diff
@@ -334,20 +385,37 @@ object JsonParser {
   /* Buffer used to parse JSON.
    * Buffer is divided to one or more segments (preallocated in Segments pool).
    */
-  private[json] class Buffer(in: Reader, closeAutomatically: Boolean) {
-    var offset = 0
+  private[json] final class Buffer(in: Reader, closeAutomatically: Boolean) {
+    // Reused by the parser when appropriate, allows for a single builder to be
+    // used throughout the parse process, and to be written to directly from the
+    // substring method, so as to avoid allocating new builders when avoidable.
+    private[json] final val builder = new java.lang.StringBuilder(32)
+
+    var offset = 0 // how far into the current segment we've read data
     var curMark = -1
     var curMarkSegment = -1
     var eofIsFailure = false
-    private[this] var segments: List[Segment] = List(Segments.apply())
+    private[this] var segments = scala.collection.mutable.ArrayBuffer(Segments.apply())
     private[this] var segment: Array[Char] = segments.head.seg
     private[this] var cur = 0 // Pointer which points current parsing location
     private[this] var curSegmentIdx = 0 // Pointer which points current segment
 
-    def mark = { curMark = cur; curMarkSegment = curSegmentIdx }
-    def back = cur = cur-1
+    // Mark the current point so that future substring calls will extract the
+    // value from this point to whatever point the buffer has advanced to.
+    final def mark = {
```

Comment:
I'll nuke these and see if it has any detrimental effect. `final` generally lets the JVM do better optimizations like inlining because it knows there's no possibility of override, but the visibility might be sufficient in this case.